lib/class-wp-rest-menu-items-controller.php

```diff
@@ -0,0 +1,658 @@
+			}
+
+			return $nav_menu_item_id;
+		}
+
+		$nav_menu_item = $this->get_nav_menu_item( $nav_menu_item_id );
+
+		/** This action is documented in wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php */
+		do_action( "rest_insert_{$this->post_type}", $nav_menu_item, $request, false );
+
+		$schema = $this->get_item_schema();
+
+		if ( ! empty( $schema['properties']['meta'] ) && isset( $request['meta'] ) ) {
+			$meta_update = $this->meta->update_value( $request['meta'], $post->ID );
+
+			if ( is_wp_error( $meta_update ) ) {
+				return $meta_update;
+			}
+		}
+
+		$nav_menu_item = $this->get_nav_menu_item( $nav_menu_item_id );
+		$fields_update = $this->update_additional_fields_for_object( $nav_menu_item, $request );
+
+		if ( is_wp_error( $fields_update ) ) {
+			return $fields_update;
+		}
+
+		$request->set_param( 'context', 'edit' );
+
+		/** This action is documented in wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php */
+		do_action( "rest_after_insert_{$this->post_type}", $nav_menu_item, $request, false );
+
+		$response = $this->prepare_item_for_response( $nav_menu_item, $request );
+
+		return rest_ensure_response( $response );
+	}
+
+	/**
+	 * Prepares a single post for create or update.
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 *
+	 * @return stdClass|WP_Error Post object or WP_Error.
+	 */
+	protected function prepare_item_for_database( $request ) {
+		$prepared_nav_item = array(
+			'menu-item-db-id'       => 0,
+			'menu-item-object-id'   => 0,
+			'menu-item-object'      => '',
+			'menu-item-parent-id'   => 0,
+			'menu-item-position'    => 0,
+			'menu-item-type'        => 'custom',
+			'menu-item-title'       => '',
+			'menu-item-url'         => '',
+			'menu-item-description' => '',
+			'menu-item-attr-title'  => '',
+			'menu-item-target'      => '',
+			'menu-item-classes'     => '',
+			'menu-item-xfn'         => '',
+			'menu-item-status'      => 'publish',
+		);
+
+		$mapping = array(
+			'menu-item-db-id'       => 'db_id',
+			'menu-item-object-id'   => 'object_id',
+			'menu-item-object'      => 'object',
+			'menu-item-parent-id'   => 'menu_item_parent',
+			'menu-item-position'    => 'menu_order',
+			'menu-item-type'        => 'type',
+			'menu-item-title'       => 'title',
+			'menu-item-url'         => 'url',
+			'menu-item-description' => 'description',
+			'menu-item-attr-title'  => 'attr_title',
+			'menu-item-target'      => 'target',
+			'menu-item-classes'     => 'classes',
+			'menu-item-xfn'         => 'xfn',
+			'menu-item-status'      => 'status',
+		);
+
+		$schema = $this->get_item_schema();
+
+		foreach ( $mapping as $original => $api_request ) {
+			if ( ! empty( $schema['properties'][ $api_request ] ) && isset( $request[ $api_request ] ) ) {
+				if ( isset( $schema['properties'][ $api_request ]['type'] ) && 'integer' === $schema['properties'][ $api_request ]['type'] ) {
+					$prepared_nav_item[ $original ] = absint( $request[ $api_request ] );
```

Comment:
This should probably go through `rest_sanitize_value_from_schema()` instead.

Reply:
I didn't know about this function. I will change this. Thanks.